### ISSUE TITLE
What is the importance of the flatten layer?

### ISSUE THREAD

**Issue by abdou31** [yinguobing/cnn-facial-landmark#81]
Hello, 
I focus on my project for now, as I see when I read the document that you gave me is that the network is like this:
![image](https://user-images.githubusercontent.com/19480228/66399158-5be19500-e9df-11e9-9061-d9873a911a83.png)

But when I see the script `landmark.py` (old version), I noticed that there is an other layer that does not write on the document, I'm talking about the flattening layer.
I would like to know is it important to mention that layer or no?

**Comment by yinguobing** [comment]
The flatten layer is required so the input of the following dense layer is a vector other than matrix.

**Comment by abdou31** [comment]
I understand you
You mean that the input of the first dense layer ( full 1024 ) should be vector.
For what reason? 
 + Is it to input the vector to tensors of the future neural network or just to the second dense layer ( full 2n ) " logits\BiasAdd" ?
